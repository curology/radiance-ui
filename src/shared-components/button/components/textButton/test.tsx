import React from 'react';
import { render } from 'src/tests/testingLibraryHelpers';

import { TextButton } from './index';

describe('<TextButton />', () => {
  describe('UI snapshots', () => {
    it('renders without any props', () => {
      const { container } = render(<TextButton>Button Text</TextButton>);

      expect(container.firstElementChild).toMatchSnapshot();
    });
    it('renders with disabled prop', () => {
      const { container } = render(
        <TextButton disabled>Disabled Button Text</TextButton>,
      );

      expect(container.firstElementChild).toMatchSnapshot();
    });
  });

  describe('onClick callback', () => {
    it('should invoke onClick', async () => {
      const spy = jest.fn();
      const { getByRole, user } = render(
        <TextButton onClick={spy}>Button Text</TextButton>,
      );

      await user.click(getByRole('button'));

      expect(spy).toHaveBeenCalled();
    });

    it('should not be clickable if disabled', async () => {
      const spy = jest.fn();
      const { getByRole, user } = render(
        <TextButton disabled onClick={spy}>
          Button Text
        </TextButton>,
      );

      await user.click(getByRole('button'));

      expect(spy).not.toHaveBeenCalled();
    });
  });
});
