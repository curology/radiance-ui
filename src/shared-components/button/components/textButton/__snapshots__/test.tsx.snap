// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextButton /> UI snapshots renders with disabled prop 1`] = `
.emotion-0 {
  border-color: transparent;
  background-color: transparent;
  border-radius: 4px;
  color: #C3C0CD;
  cursor: not-allowed;
}

.emotion-0:hover {
  opacity: 0.8;
  background-color: transparent;
}

.emotion-0:active,
.emotion-0:focus {
  outline: none;
  box-shadow: 0px 0px 0px 2px #FFFFFF,0px 0px 0px 4px #332E54;
}

.emotion-2 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100%;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  padding: 1rem 0;
  -webkit-transition: -webkit-transform 350ms;
  transition: transform 350ms;
  width: 100%;
  -webkit-transform: translateX(0);
  -moz-transform: translateX(0);
  -ms-transform: translateX(0);
  transform: translateX(0);
}

.emotion-2>svg {
  opacity: 1;
  -webkit-transition: opacity 350ms;
  transition: opacity 350ms;
  margin-right: 1rem;
  margin-top: -5px;
}

<button
  class="emotion-0 emotion-1"
  disabled=""
>
  <div
    class="emotion-2 emotion-3"
  >
    Disabled Button Text
  </div>
</button>
`;

exports[`<TextButton /> UI snapshots renders without any props 1`] = `
.emotion-0 {
  border-color: transparent;
  background-color: transparent;
  border-radius: 4px;
  color: #332E54;
  cursor: pointer;
}

.emotion-0:hover {
  opacity: 0.8;
  background-color: transparent;
}

.emotion-0:active,
.emotion-0:focus {
  outline: none;
  box-shadow: 0px 0px 0px 2px #FFFFFF,0px 0px 0px 4px #332E54;
}

.emotion-2 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100%;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  padding: 1rem 0;
  -webkit-transition: -webkit-transform 350ms;
  transition: transform 350ms;
  width: 100%;
  -webkit-transform: translateX(0);
  -moz-transform: translateX(0);
  -ms-transform: translateX(0);
  transform: translateX(0);
}

.emotion-2>svg {
  opacity: 1;
  -webkit-transition: opacity 350ms;
  transition: opacity 350ms;
  margin-right: 1rem;
  margin-top: -5px;
}

<button
  class="emotion-0 emotion-1"
>
  <div
    class="emotion-2 emotion-3"
  >
    Button Text
  </div>
</button>
`;
